pagefaultgames/pokerogue
# [Refactor] Pokerogue API client

## What are the changes the user will see?

n/a

## Why am I making these changes?

Well you know how it is... the way it was designed in the code was bugging @Opaque02  & me.  I thought I fast and easy fix it. One thing leads to another and here we are..

> [!WARNING]
> **I couldn't check the discord/google unlink requests due to local limitations.**
> 
> *Yes, I did set up my own Discord Dev Application to link it to but it simply refused to save those discord IDs into the database. No idea why*
> <img width="200" alt="image" src="https://github.com/user-attachments/assets/783027b3-9c40-4ba7-af8b-a60f048c169a">


## What are the changes from a developer perspective?

All `Utils.apiFetch` and `Utils.apiPost` were replaced with a new `PokerogueApi` class which is structured into sub-apis

Every request & response from the API is properly typed now

### Screenshots/Videos

n/a

## How to test the changes?

1. Make sure you have the [rogueserver repo](https://github.com/pagefaultgames/rogueserver/) cloned
2. Start the local server
3. Make sure to set `VITE_BYPASS_LOGIN` to `0` in `env.development`
4. Start the client
5. Tests all the things that cause requests 


## Checklist
- [x] Have I considered writing automated tests for the issue? => there is some already
- ~~[] Are the changes visual?~~
